enclave_manager/Dockerfile

```diff
@@ -0,0 +1,214 @@
+    if ([ -c /dev/isgx ] && [ -S /var/run/aesmd/aesm.socket ]); then SGX_MODE=HW; \
+    else SGX_MODE=SIM; \
+    fi; \
+    make SGX_MODE=${SGX_MODE} DESTDIR=/opt/intel/sgxssl all test ) \
+ && (cd intel-sgx-ssl/Linux; make install ) \
+ && rm -rf /tmp/intel-sgx-ssl \
+ && echo "SGX_SSL=/opt/intel/sgxssl" >> /etc/environment
+
+COPY . /project/avalon
+
+ARG SGX_MODE
+ARG MAKECLEAN
+ARG TCF_DEBUG_BUILD
+
+# Environment setup
+
+RUN echo "TCF_HOME=/project/avalon/" >> /etc/environment \
+ && if [ ! -z "$http_proxy"  ]; then \
+    echo 'Acquire::http::Proxy "'$http_proxy'";' >> /etc/apt/apt.conf.d/00proxy; \
+        echo "http_proxy=$http_proxy" >> /etc/wgetrc; \
+        echo "http_proxy=$http_proxy" >> /etc/environment; \
+        echo "HTTP_PROXY=$(echo $http_proxy | sed 's,[a-zA-Z]*://,,')" >> /etc/environment; \
+    fi \
+ && if [ ! -z "$ftp_proxy"  ];  then \
+    echo 'Acquire::ftp::Proxy "'$ftp_proxy'";' >> /etc/apt/apt.conf.d/00proxy; \
+        echo "ftp_proxy=$ftp_proxy" >> /etc/wgetrc; \
+        echo "ftp_proxy=$ftp_proxy" >> /etc/environment; \
+    fi \
+ && if [ ! -z "$https_proxy" ]; then \
+    echo 'Acquire::https::Proxy "'$https_proxy'";' >> /etc/apt/apt.conf.d/00proxy; \
+        echo "https_proxy=$https_proxy" >> /etc/wgetrc; \
+        echo "https_proxy=$https_proxy" >> /etc/environment; \
+        echo "HTTPS_PROXY=$(echo $https_proxy | sed 's,[a-zA-Z]*://,,')" >> /etc/environment; \
+    fi
+
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
+ENV TCF_HOME=/project/avalon
+ENV SGX_SSL=/opt/intel/sgxssl
+ENV SGX_SDK=/opt/intel/sgxsdk
+ENV PATH=$PATH:$SGX_SDK/bin:$SGX_SDK/bin/x64
+ENV PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$SGX_SDK/pkgconfig
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$SGX_SDK/sdk_libs
+ENV TCF_ENCLAVE_CODE_SIGN_PEM="$TCF_HOME/enclave.pem"
+
+RUN openssl genrsa -3 -out $TCF_HOME/enclave.pem 3072
+
+WORKDIR /project/avalon/common/sgx_workload
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/examples/apps
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/common/cpp
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/tc/sgx/trusted_worker_manager/common
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/tc/sgx/trusted_worker_manager/enclave
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/common/python
+
+RUN make \
+  && make install
+
+#ENV PATH=$PATH:/project/avalon/examples/enclave_manager
+
+WORKDIR /project/avalon/sdk
+
+RUN python3 setup.py bdist_wheel && pip3 install dist/*.whl
+
+WORKDIR /project/avalon/enclave_manager
+
+RUN echo "Building avalon enclave manager" \
+  && make \
+  && make install
```

Comment:
Try to follow multistage Dockerfile in order to reduce the size of final docker image. You can follow https://github.com/hyperledger/avalon/blob/1b7422804c768eabec0635bba564bd4e6803b3a4/shared_kv_storage/Dockerfile or https://github.com/hyperledger/avalon/blob/master/listener/Dockerfile